recipes/newrelic/apm/php/debian.yml

```diff
@@ -241,38 +241,35 @@ install:
               continue
             fi
             priv_user=$(ps auxf | egrep $full_process_name | grep -v grep | grep -v yml | head -n1 | awk '{print $1}')
-            fpm_nginx_process_name=$(echo "${full_process_name}" | sed -n 's/\(.*\):$/\1/p')
-            if [ -n "${fpm_nginx_process_name}" ]; then
-              fpm_process_loc=$(ps auxf | egrep $full_process_name | grep -v grep | grep -v yml | head -n1 | awk '{print $14}')
-              fpm_ver=$(echo "${fpm_process_loc}" | sed -n 's/.*\/php\/\([578].[0-9]\)\/.*/\1/p')
-              if [ -n "${fpm_ver}" ]; then
-                # It's fpm.
-                process_name="php${fpm_ver}-fpm"
-              else
-                # It's nginx.
-                process_name=$fpm_nginx_process_name
-              fi
-            else
-              process_name=$(echo $full_process_name | sed -n 's/.*\/\(.*\)$/\1/p')
+
+            pid=$(ps auxf | egrep $full_process_name | grep -v grep | grep -v yml | head -n1 | awk '{print $2}')
+            # Use the pid to find the systemctl service name
+            service_name=$(systemctl status $pid | head -n1 | awk '{print $2}')
+            if [ ${#service_name} -le 8 ]
+            then
+              echo -e "{{.RED}}Found php process \"${full_process_name}\" but failed to find associated systemctl service{{.NOCOLOR}}"
+              exit 131
```

Comment:
Should this be a `continue` or an `exit`?

Reply:
First I don't know I'd use the phrasing "Found php process. .." since this section is looking for other services - maybe say "Found service process ...".
I would lean towards an exit since the recipe has encountered an unexpected condition.